lib/rucio/core/did.py

```diff
@@ -577,7 +577,8 @@ def attach_dids_to_dids(attachments, account, ignore_duplicate=False, session=No
         except NoResultFound:
             raise exception.DataIdentifierNotFound("Data identifier '%s:%s' not found" % (attachment['scope'], attachment['name']))
 
-        session.bulk_insert_mappings(models.UpdatedDID, parent_dids)
+    parent_dids = [dict(tup) for tup in set(tuple(dictionary.items()) for dictionary in parent_dids)]
```

Comment:
oof, please add comment what this does :-)

Reply:
You're right. I thought I implemented something to skip the insertion if all the files are duplicate, but it doesn't seem to be the case. Will do it.